src/buildinfo.rs

```diff
@@ -24,93 +31,130 @@ pub struct BuildInfo<I, O> {
 
 impl<I: DeserializeOwned, O: DeserializeOwned> BuildInfo<I, O> {
     /// Deserializes the `BuildInfo` object from the given file
-    pub fn read(path: impl AsRef<Path>) -> Result<Self, SolcError> {
+    pub fn read(path: impl AsRef<Path>) -> Result<Self> {
         utils::read_json_file(path)
     }
 }
 
+/// Additional context we cache for each compiler run.
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct BuildContext<L> {
+    /// Mapping from internal compiler source id to path of the source file.
+    pub source_id_to_path: HashMap<u32, PathBuf>,
```

Comment:
is this part of the cache file? if so I'd prefer btreemap so it's consistent
```suggestion
    pub source_id_to_path: BTreeMap<u32, PathBuf>,
```